### ISSUE TITLE
an asset file is handled by two workers

### ISSUE THREAD

**Issue by teazean** [parcel-bundler/parcel#946]
**Choose one:** a 🐛 bug report 

### 🎛 Configuration (.babelrc, package.json, cli command)

```bash
parcel build index.html --public-url ./ --no-cache
```

```
.
├── index.html
└── main.js
```
index.html
```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Document</title>
</head>
<body>
    <script src="main.js"></script>
</body>
</html>
```
main.js
```javscript
console.log(1)
```

### 🤔 Expected Behavior

`main.js` should only be processed once.

### 😯 Current Behavior
put ``console.log(`run worker : ${path}\n`);`` below this [line](https://github.com/parcel-bundler/parcel/blob/1ceb7aa791cc4a4a2291c1f78c799182f89fc111/src/worker.js#L16)

As below shows, there are two workers handling the `main.js`:
<img width="537" alt="bash" src="https://user-images.githubusercontent.com/4465253/36963646-0a5d7340-208f-11e8-8a78-ee80fc12ee28.png">

### Possible Reason

when deal with `main.js` , there are [two steps](https://github.com/parcel-bundler/parcel/blob/1ceb7aa791cc4a4a2291c1f78c799182f89fc111/src/Bundler.js#L414): 
1. `resolveDep(main.js)` -> `resolveAsset(main.js)` -> `getAsset(main.js)` -> ~~`buildQueue.add(main.js)` -> `new worker(main.js)`~~
2. `loadAsset(main.js)` -> `wokerFarm.run(main.js)` -> `new worker(main.js)`

As shown above, am asset named `main.js` leads two workers.

### 💁 Possible Solution

we can run then existing asset not asset path. 

### 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 1.6.2 |
| Node             | 8.3.0 |
| npm/Yarn         | npm |
| Operating System | macOS 10.13.3 |

**Comment by DeMoorJasper** [comment]
This is expected behaviour as implemented here: #567
This causes the JIT compiler to preload the dependencies of parcel inside the workers before actually using them speeding up the process quite a lot on small projects (while not impacting large projects)

**Comment by teazean** [comment]
wonderful.
